crates/autopilot/src/decoded_settlement.rs

```diff
@@ -179,41 +179,29 @@ impl From<(Address, U256, Bytes<Vec<u8>>)> for DecodedInteraction {
 #[derive(Debug, Clone)]
 pub struct OrderExecution {
     pub order_uid: OrderUid,
-    pub executed_solver_fee: Option<U256>,
-    pub sell_token: H160,
-    pub buy_token: H160,
-    pub sell_amount: U256,
-    pub buy_amount: U256,
-    pub executed_amount: U256,
     pub signature: Vec<u8>, //encoded signature
-    // For limit orders the solver computes the fee
-    pub solver_determines_fee: bool,
     pub owner: H160,
+    pub executed_amount: U256,
+    pub executed_fee: ExecutedFee,
 }
 
-impl TryFrom<database::orders::OrderExecution> for OrderExecution {
-    type Error = anyhow::Error;
-
-    fn try_from(order: database::orders::OrderExecution) -> std::result::Result<Self, Self::Error> {
-        let owner = H160(order.owner.0);
+impl OrderExecution {
+    pub fn new(order: &Order, execution: database::orders::OrderExecution) -> Result<Self> {
+        let owner = H160(execution.owner.0);
         Ok(Self {
-            order_uid: OrderUid(order.order_uid.0),
-            executed_solver_fee: order
-                .executed_solver_fee
-                .as_ref()
-                .and_then(big_decimal_to_u256),
-            sell_token: H160(order.sell_token.0),
-            buy_token: H160(order.buy_token.0),
-            sell_amount: big_decimal_to_u256(&order.sell_amount).context("sell_amount")?,
-            buy_amount: big_decimal_to_u256(&order.buy_amount).context("buy_amount")?,
-            executed_amount: big_decimal_to_u256(&order.executed_amount).unwrap(),
+            order_uid: OrderUid(execution.order_uid.0),
             signature: {
-                let signing_scheme = signing_scheme_from(order.signing_scheme);
-                let signature = Signature::from_bytes(signing_scheme, &order.signature)?;
+                let signing_scheme = signing_scheme_from(execution.signing_scheme);
+                let signature = Signature::from_bytes(signing_scheme, &execution.signature)?;
                 signature.encode_for_settlement(owner).to_vec()
             },
-            solver_determines_fee: order.class == OrderClass::Limit,
             owner,
+            executed_amount: big_decimal_to_u256(&execution.executed_amount).unwrap(),
+            executed_fee: if order.metadata.solver_fee == U256::zero() {
```

Comment:
Wouldn't it be more correct to check `order.data.fee_amount` here? solver_fee may be scaled (so it's not the actual fee paid) and is something we want to remove anyway, so why not try to not use it wherever possible?

Technically we don't even need to fetch the order from the database, the fee amount can also be recovered from the on-chain event.

Reply:
I see, where in the rewards calculation do we still use the solver fee? To compute the observed score?
In this case I agree we need to keep it as is.

Reply:
Yes, in [this script](https://github.com/cowprotocol/solver-rewards/blob/main/queries/orderbook/batch_rewards.sql) we refer to the `fee`, where for the calculation of the `fee` we use the unsubsidized value.